### DISCUSSION TITLE
コマンドの作り方について

### DISCUSSION THREAD

**Discussion by masada-tea** [HANON-games/Shiden#18] [Q&A] [answered]
いつもお疲れ様です。
コマンドの作り方について試行錯誤しています。
初心者なもので自信がないので例えばなのですが、
1.マップ上にplayer、幾つかのオブジェクト（例：椅子、箪笥、壺）を配置
2.コマンドを実行するとplayerが指定したオブジェクトのある位置に移動する
という感じのコマンドを作りたいと思っています。
（もしかしたらカメラをそれぞれの前に配置して切り替えるコマンドの方がいいのかもしれませんが）
この場合、コマンドはどのように組むべきでしょうか？
どうもlocation関係のものがブループリント上で選べずやり方に詰まっています。
よろしくお願いします。

**Comment by HANON-games** [comment, marked as answer]
@masada-tea 
こんにちは！
ShidenCommand は UObject を直に継承しているため、使用できるノードには制限があります。
コマンドの Blueprint で使用できない機能を使いたい場合、Widget か Shiden のシナリオを再生している Actor のほうに処理を実装し、そちらをコマンドから呼び出してください。
![image](https://github.com/user-attachments/assets/736f8d09-29ac-4fc4-860b-feb3926f7e0a)
